### PR TITLE
Upsamp2

### PR DESCRIPTION
**Pull request by erakra**
handling fractional bilinear upsampling

### CODE REVIEW

**Comment by slefrancois** [comment]
fixes #5995
jenkins test this win mac

**Comment by slefrancois** [comment]
jenkins test win mac

**Comment by nouiz** [comment]
all CI "passed" One failure have been fixed in master and whe other was a cancellation.

File: theano/tensor/nnet/abstract_conv.py [hunk 1]
```diff
@@ -1751,13 +1760,114 @@ def bilinear_kernel_1D(ratio, normalize=True):
     return kern
 
 
+def frac_bilinear_upsampling(input,
+                             ratio=None,
```
Comment (lamblin) [inline theano/tensor/nnet/abstract_conv.py]: I think you can simplify this a bit by removing the `ratio` parameter.
The only current codepath that calls `frac_bilinear_upsampling` (l. 1920) is only taken if `frac_ratio` is provided and `ratio` is not.
If someone really wants to call `frac_bilinear_upsampling` manually, I think it is OK to ask for them to provide `frac_ratio=(ratio, 1)`.

**Comment by slefrancois** [comment]
jenkins test this win

**Comment by slefrancois** [comment]
jenkins test win

**Comment by lamblin** [comment]
I'm merging now, the simplification can be done later.

**Pull request merged**